### DISCUSSION TITLE
Automatically sync all workspaces using VCS

### DISCUSSION THREAD

**Discussion by TahirJalilov** [Kong/insomnia#3769] [Ideas]
Hello. Am I correct to understand that VCS not automatically syncs all my workspaces? And every time after adding a new folder or request to the workspace I have to create a snapshot and push it to the VCS?

**Comment by ravindu1024** [comment]
Ah just came here to suggest the same thing. Would be nice to have an auto "sync all" feature or maybe an option to include other workspaces when creating the snapshot.

**Comment by DMarby** [comment]
Hello @TahirJalilov, @ravindu1024!

We are looking into having the VCS automatically sync changes in the future.
One of the issues with the automatic legacy sync system that we'd like to fix when doing so was that it would silently ignore merge conflicts (when something has changed both remotely and locally), what would be your preferred behaviour in such a case for you?

For example, would you prefer it to just pick the most recent change, or get a notification and manually resolve the conflict by picking which version you'd prefer, or something else entirely?

**Comment by skiat** [comment]
> 
> 
> Hello @TahirJalilov, @ravindu1024!
> 
> We are looking into having the VCS automatically sync changes in the future.
> One of the issues with the automatic legacy sync system that we'd like to fix when doing so was that it would silently ignore merge conflicts (when something has changed both remotely and locally), what would be your preferred behaviour in such a case for you?
> 
> For example, would you prefer it to just pick the most recent change, or get a notification and manually resolve the conflict by picking which version you'd prefer, or something else entirely?

Being now used to the old "the last push wins" I'd tend to say that this way one can yes break things.. but another can also repair it once for everyone. 

But after better reflections of course a notification about conflicts & what pick option, will be better!
The important I think is to restore the auto-sync 👍

**Comment by boardy** [comment]
Hi, 
Agreed, that I'm missing auto sync feature, I move between couple of PCs quite often and I've lost count the number of times that I've made a request on one machine and so used to the previous sync system, I've forgotten to push the changes to VCS until I get to my other PC and find I'm missing loads of requests. 

Would agree that the most recent change is very likely to be the correct case, 9 times out of 10, and on the odd time that it might not be, its likely going to be simple to resolve manually. 

Thanks

**Comment by lacherogwu** [comment]
I don't get why you even did this change
the previous system was perfect.

I log in to the same account on different computers
if I change something in one place, I should see it in the other place.

it's my personal account I don't understand why I need to create snapshots and push code, that's ridiculous when a user works alone

This branching system is probably very necessary when working with teams, but as a sole it is just annoying
how many times I forgot to create a snapshot and push
and not just forget, it is simply not marking any sense that on every change I make I need to create a snapshot and push just for me to have it available across my account in different places...

I hope you have a solution for this because Insomnia was my favorite tool, but after this change, it makes my life harder and I think to look for an alternative.
